analyzer/pom-analyzer/src/main/java/eu/fasten/analyzer/pomanalyzer/POMAnalyzerPlugin.java

```diff
@@ -262,6 +271,29 @@ public long saveToDatabase(String product, String version, String repoUrl, Strin
             return packageVersionId;
         }
 
+        private void addDependenciesToWorkingSet(MetadataDao metadataDao) {
+        	
+        	for (var dep : dependencyData.dependencies) {
+        		var depProduct = dep.groupId + Constants.mvnCoordinateSeparator + dep.artifactId;
+        		
+	            var depVersions = dep.getVersion().split(Pattern.quote(","));
+	            for (String v : depVersions) {
+	                if (MavenUtilities.mavenArtifactExists(dep.groupId, dep.artifactId, v, null) &&
+	                        (metadataDao.getPackageVersion(depProduct, v) == null ||
+	                                !metadataDao.isArtifactIngested(depProduct, v))) {
```

Comment:
This check should be removed - either it will unnecessarily delay dependencies from being processed, or it is redundant if the package-version has already been added to the DB (because then the other clause will be true).